reporter/src/main/kotlin/reporters/OpossumReporter.kt

```diff
@@ -0,0 +1,599 @@
+                },
+                "attributionBreakpoints" to attributionBreakpoints,
+                "filesWithChildren" to filesWithChildren,
+                "frequentLicenses" to frequentLicenses.toList().map { it.toJson() },
+                "baseUrlsForSources" to baseUrlsForSources,
+                "externalAttributionSources" to externalAttributionSources
+            )
+
+        fun getSignalsForFile(file: String): List<OpossumSignal> =
+            pathToSignal[file]
+                ?.mapNotNull { uuid -> signals.find { it.uuid == uuid } }
+                ?: emptyList()
+
+        fun addAttributionBreakpoint(breakpoint: String) {
+            attributionBreakpoints.add(resolvePath(breakpoint, isDirectory = true))
+            resources.addResource(breakpoint)
+        }
+
+        fun addFileWithChildren(fileWithChildren: String) {
+            filesWithChildren.add(resolvePath(fileWithChildren, isDirectory = true))
+        }
+
+        fun addBaseURL(path: String, vcs: VcsInfo) {
+            val idFromPath = resolvePath(path, isDirectory = true)
+
+            if (baseUrlsForSources.containsKey(idFromPath)) { return }
+
+            if (vcs.type == VcsType.GIT && (vcs.url.startsWith("https://github.com/")
+                        || vcs.url.startsWith("ssh://git@github.com/"))
+            ) {
+                val revision = vcs.revision.takeIf { it.isNotBlank() } ?: "HEAD"
+                val baseUrl = vcs.url
+                    .replace("ssh://git@github.com/", "https://github.com/")
+                    .replace(Regex("\\.git$"), "")
+                    .plus("/tree/$revision/${vcs.path}/{path}")
+                baseUrlsForSources[idFromPath] = baseUrl
+            }
+        }
+
+        fun addExternalAttributionSource(key: String, name: String, priority: Int): String {
+            if (!externalAttributionSources.containsKey(key)) {
+                externalAttributionSources[key] = OpossumExternalAttributionSource(name, priority)
+            }
+            return key
+        }
+
+        fun addPackageRoot(id: Identifier, path: String, level: Int = 0, vcs: VcsInfo = VcsInfo.EMPTY) {
+            val mapOfId = packageToRoot.getOrPut(id) { sortedMapOf() }
+            val oldLevel = mapOfId.getOrDefault(path, level)
+            mapOfId[path] = min(level, oldLevel)
+            resources.addResource(path)
+
+            addBaseURL(path, vcs)
+        }
+
+        fun addSignal(signal: OpossumSignal, paths: SortedSet<String>) {
+            if (paths.isEmpty()) { return }
+
+            val matchingSignal = signals.find { it.matches(signal) }
+            val uuidOfSignal = if (matchingSignal == null) {
+                signals.add(signal)
+                signal.uuid
+            } else {
+                matchingSignal.uuid
+            }
+
+            paths.forEach {
+                log.trace("add signal ${signal.id} of source ${signal.source} to $it")
```

Comment:
Please use lambda-arguments for logging: `log.trace { ... }`. Similar in more places.